packages/fxa-payments-server/src/components/CouponForm/index.tsx

```diff
@@ -0,0 +1,106 @@
+import React, { useState, MouseEventHandler, FormEventHandler } from 'react';
+import { Localized } from '@fluent/react';
+import './index.scss';
+import { Coupon } from '../../lib/Coupon';
+
+type CouponFormProps = {
+  coupon?: Coupon;
+  setCoupon: (coupon: Coupon | undefined) => void;
+};
+
+export const CouponForm = ({ coupon, setCoupon }: CouponFormProps) => {
+  const [hasCoupon, setHasCoupon] = useState(coupon ? true : false);
+  const [couponCode, setCouponCode] = useState(coupon ? 'test' : '');
+  const [error, setError] = useState(false);
+
+  const onSubmit: FormEventHandler = (event: any) => {
+    event.preventDefault();
+    event.stopPropagation();
+    if (couponCode === 'test') {
+      setHasCoupon(true);
+      setCoupon({
+        amount: 200,
+      });
+    } else {
+      setError(true);
+    }
+  };
+
+  const removeCoupon: MouseEventHandler<HTMLButtonElement> = (event: any) => {
+    event.preventDefault();
+    event.stopPropagation();
+    setHasCoupon(false);
+    setCoupon(undefined);
+  };
+
+  return (
+    <section
+      className={`coupon-component ${coupon ? 'has-coupon' : ''}`}
+      data-testid="coupon-component"
+    >
+      <div className="coupon-component-inner">
+        {hasCoupon ? (
```

Comment:
Why two of these?  Seems possible to combine them?

Reply:
i was following the pattern of the plan details component. probably possible to combine them but at the expense of refactoring the css a bit - this could possible be spun off into a separate ticket for a maintanence sprint where we do a bit of refactoring on templates/css